File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Java/service/OrderService.java
package service;

import models.Order;
import utils.OrderStatus;

import java.util.HashMap;
import java.util.Map;

public class OrderService {
    private Map<String, Order> orders = new HashMap<>();

    public void placeOrder(Order order) {
        orders.put(order.getId(), order);
    }

    public Order getOrderById(String id) {
        return orders.get(id);
    }

    public void updateOrderStatus(String id, OrderStatus status) {
        Order order = orders.get(id);
        if (order != null) {
            order.setStatus(status);
        }
    }
}

File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Cpp/service/src/OrderService.cpp
#include "../include/OrderService.h"

void OrderService::placeOrder(const std::shared_ptr<Order>& order) {
    orders[order->getId()] = order;
}

std::shared_ptr<Order> OrderService::getOrderById(const std::string& id) const {
    auto it = orders.find(id);
    if (it != orders.end()) {
        return it->second;
    }
    return nullptr;
}

void OrderService::updateOrderStatus(const std::string& id, OrderStatus status) {
    auto it = orders.find(id);
    if (it != orders.end() && it->second != nullptr) {
        it->second->setStatus(status);
    }
}
